docs: add cli steps for org sync

EdwardAngert opened the pull request:
[preview](https://coder.com/docs/@15431-docs-org-sync/admin/users/idp-sync#organization-sync-premium)

Commit:
add cli steps for org sync

Commit:
dashboard steps to steps

Commit:
html tags work better when they're closed

Commit:
md spacing

Commit:
md and steps cleanup

Commit:
Merge remote-tracking branch 'origin' into 15431-docs-org-sync

Commit:
Merge remote-tracking branch 'origin/main' into 15431-docs-org-sync

Emyrk commented on docs/admin/users/idp-sync.md:
```diff
@@ -304,77 +339,141 @@ Visit the Coder UI to confirm these changes:
 
-Next configure the Coder server to read groups from the claim name with the OIDC
-organization field server flag:
+### CLI
```
Can we default to the `Dashboard` view? The cli is going to be more error prone and likely used after they understand the feature (via dashboard mucking around)

Emyrk commented on docs/admin/users/idp-sync.md:
```diff
@@ -304,77 +339,141 @@ Visit the Coder UI to confirm these changes:
+   | mapping                     | Mapping takes a claim from the IdP, and associates it with 1 or more organizations by UUID. </br> No validation is done, so you can put UUID's of orgs that do not exist (a noop). The UI picker will allow selecting orgs from a drop down, and convert it to a UUID for you.          |
+   | organization_assign_default | This setting exists for maintaining backwards compatibility with single org deployments, either through their upgrade, or in perpetuity. </br> If this is set to 'true', all users will always be assigned to the default organization regardless of the mappings and their IdP claims. |
+
+### Dashboard
```
```suggestion
### Enviroment Config
```

This reads `dashboard`, but these docs are configuring via env vars. We should change this title, and have some warning like **NOT RECOMMENDED** and **will be overwritten by database settings**.

We do need a `Dashboard` section, and that section should be the default. @jaaydenh just recently finished the UI.

EdwardAngert replied:
@Emyrk I merged in the Dashboard PR, so these steps are now updated - that includes removing the env vars steps

EdwardAngert replied:
done 

![2024-12-10_10-48-29](https://github.com/user-attachments/assets/480ffc94-d435-4439-bac4-84b77733f722)

Emyrk reviewed (changes requested):
Content is good, just need to change section headings, and add a new `Dashboard` section, making it the default (first).

EdwardAngert committed:
docs: add new steps for org sync through ui (#15768)

[preview](https://coder.com/docs/@15503-ui-org-sync/admin/users/idp-sync#organization-sync-premium)

---------

Co-authored-by: EdwardAngert <2408959-EdwardAngert@users.noreply.gitlab.com>
Co-authored-by: EdwardAngert <17991901+EdwardAngert@users.noreply.github.com>

EdwardAngert committed:
dashboard tab to default
